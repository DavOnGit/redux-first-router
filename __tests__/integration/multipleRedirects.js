import createTest from '../../__helpers__/createTest'

createTest('multiple redirects are honored', {
  SECOND: {
    path: '/second',
    beforeEnter: ({ dispatch }) => {
      return dispatch({ type: 'REDIRECTED' })
    },
    thunk: function() {}
  },
  REDIRECTED: {
    path: '/redirected',
    beforeEnter: ({ dispatch }) => {
      return dispatch({ type: 'REDIRECTED_AGAIN' })
    },
    thunk: function() {}
  },
  REDIRECTED_AGAIN: {
    path: '/redirected-again',
    thunk: function() {}
  }
})

createTest('multiple redirects are honored after enter', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => {
      return dispatch({ type: 'REDIRECTED_AFTER' })
    },
    onComplete: function() {}
  },
  REDIRECTED_AFTER: {
    path: '/redirected-after',
    thunk: ({ dispatch }) => {
      return dispatch({ type: 'REDIRECTED_AGAIN_AFTER' })
    },
    onComplete: function() {}
  },
  REDIRECTED_AGAIN_AFTER: {
    path: '/redirected-again-after',
    onComplete: function() {}
  }
})
