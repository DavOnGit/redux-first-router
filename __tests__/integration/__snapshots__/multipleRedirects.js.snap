// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple redirects are honored after enter: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/redirected-again-after",
      "search": "",
      "state": Object {},
      "url": "/redirected-again-after",
    },
    "params": Object {},
    "pathname": "/redirected-again-after",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "REDIRECTED_AGAIN_AFTER",
    "url": "/redirected-again-after",
  },
]
`;

exports[`multiple redirects are honored after enter: 1 - history_index 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/redirected-again-after",
          "search": "",
          "state": Object {},
          "url": "/redirected-again-after",
        },
        "params": Object {},
        "pathname": "/redirected-again-after",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "REDIRECTED_AGAIN_AFTER",
        "url": "/redirected-again-after",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "direction": "forward",
        "entries": Array [
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/first",
              "search": "",
              "state": Object {},
              "url": "/first",
            },
            "params": Object {},
            "pathname": "/first",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "FIRST",
            "url": "/first",
          },
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/redirected-after",
              "search": "",
              "state": Object {},
              "url": "/redirected-after",
            },
            "params": Object {},
            "pathname": "/redirected-after",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "REDIRECTED_AFTER",
            "url": "/redirected-after",
          },
        ],
        "index": 1,
        "kind": "replace",
        "length": 2,
        "pathname": "/redirected-after",
        "pop": false,
        "scene": "",
        "search": "",
        "status": 302,
        "url": "/redirected-after",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "REDIRECTED_AFTER",
    },
    "index": 1,
    "kind": "replace",
    "length": 2,
    "pathname": "/redirected-again-after",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "scene": "",
    "search": "",
    "status": 302,
    "url": "/redirected-again-after",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "REDIRECTED_AGAIN_AFTER",
}
`;

exports[`multiple redirects are honored after enter: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AFTER - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AFTER - thunk 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AGAIN_AFTER - onComplete 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - routes - SECOND - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/redirected-again-after",
          "search": "",
          "state": Object {},
          "url": "/redirected-again-after",
        },
        "params": Object {},
        "pathname": "/redirected-again-after",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "REDIRECTED_AGAIN_AFTER",
        "url": "/redirected-again-after",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "direction": "forward",
        "entries": Array [
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/first",
              "search": "",
              "state": Object {},
              "url": "/first",
            },
            "params": Object {},
            "pathname": "/first",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "FIRST",
            "url": "/first",
          },
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/redirected-after",
              "search": "",
              "state": Object {},
              "url": "/redirected-after",
            },
            "params": Object {},
            "pathname": "/redirected-after",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "REDIRECTED_AFTER",
            "url": "/redirected-after",
          },
        ],
        "index": 1,
        "kind": "replace",
        "length": 2,
        "pathname": "/redirected-after",
        "pop": false,
        "scene": "",
        "search": "",
        "status": 302,
        "url": "/redirected-after",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "REDIRECTED_AFTER",
    },
    "hash": "",
    "index": 1,
    "kind": "replace",
    "length": 2,
    "params": Object {},
    "pathname": "/redirected-again-after",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED_AGAIN_AFTER",
    "universal": false,
    "url": "/redirected-again-after",
  },
  "title": "REDIRECTED_AGAIN_AFTER",
}
`;

exports[`multiple redirects are honored after enter: 1 - title 1`] = `"REDIRECTED_AGAIN_AFTER"`;

exports[`multiple redirects are honored: 1 - history_entries 1`] = `
Array [
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "FIRST",
    "url": "/first",
  },
  Object {
    "basename": "",
    "hash": "",
    "key": "345678",
    "location": Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/redirected-again",
      "search": "",
      "state": Object {},
      "url": "/redirected-again",
    },
    "params": Object {},
    "pathname": "/redirected-again",
    "query": Object {},
    "search": "",
    "state": Object {},
    "type": "REDIRECTED_AGAIN",
    "url": "/redirected-again",
  },
]
`;

exports[`multiple redirects are honored: 1 - history_index 1`] = `1`;

exports[`multiple redirects are honored: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/redirected-again",
          "search": "",
          "state": Object {},
          "url": "/redirected-again",
        },
        "params": Object {},
        "pathname": "/redirected-again",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "REDIRECTED_AGAIN",
        "url": "/redirected-again",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "direction": "forward",
        "entries": Array [
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/first",
              "search": "",
              "state": Object {},
              "url": "/first",
            },
            "params": Object {},
            "pathname": "/first",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "FIRST",
            "url": "/first",
          },
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/redirected",
              "search": "",
              "state": Object {},
              "url": "/redirected",
            },
            "params": Object {},
            "pathname": "/redirected",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "REDIRECTED",
            "url": "/redirected",
          },
        ],
        "index": 1,
        "kind": "replace",
        "length": 2,
        "pathname": "/redirected",
        "pop": false,
        "scene": "",
        "search": "",
        "status": 302,
        "url": "/redirected",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "REDIRECTED",
    },
    "index": 1,
    "kind": "replace",
    "length": 2,
    "pathname": "/redirected-again",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "scene": "",
    "search": "",
    "status": 302,
    "url": "/redirected-again",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "REDIRECTED_AGAIN",
}
`;

exports[`multiple redirects are honored: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED - beforeEnter 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED_AGAIN - thunk 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - SECOND - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/first",
          "search": "",
          "state": Object {},
          "url": "/first",
        },
        "params": Object {},
        "pathname": "/first",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "FIRST",
        "url": "/first",
      },
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "location": Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "pathname": "/redirected-again",
          "search": "",
          "state": Object {},
          "url": "/redirected-again",
        },
        "params": Object {},
        "pathname": "/redirected-again",
        "query": Object {},
        "search": "",
        "state": Object {},
        "type": "REDIRECTED_AGAIN",
        "url": "/redirected-again",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "direction": "forward",
        "entries": Array [
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/first",
              "search": "",
              "state": Object {},
              "url": "/first",
            },
            "params": Object {},
            "pathname": "/first",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "FIRST",
            "url": "/first",
          },
          Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "location": Object {
              "basename": "",
              "hash": "",
              "key": "345678",
              "pathname": "/redirected",
              "search": "",
              "state": Object {},
              "url": "/redirected",
            },
            "params": Object {},
            "pathname": "/redirected",
            "query": Object {},
            "search": "",
            "state": Object {},
            "type": "REDIRECTED",
            "url": "/redirected",
          },
        ],
        "index": 1,
        "kind": "replace",
        "length": 2,
        "pathname": "/redirected",
        "pop": false,
        "scene": "",
        "search": "",
        "status": 302,
        "url": "/redirected",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "REDIRECTED",
    },
    "hash": "",
    "index": 1,
    "kind": "replace",
    "length": 2,
    "params": Object {},
    "pathname": "/redirected-again",
    "pop": false,
    "prev": Object {
      "basename": "",
      "direction": "forward",
      "entries": Array [
        Object {
          "basename": "",
          "hash": "",
          "key": "345678",
          "location": Object {
            "basename": "",
            "hash": "",
            "key": "345678",
            "pathname": "/first",
            "search": "",
            "state": Object {},
            "url": "/first",
          },
          "params": Object {},
          "pathname": "/first",
          "query": Object {},
          "search": "",
          "state": Object {},
          "type": "FIRST",
          "url": "/first",
        },
      ],
      "hash": "",
      "index": 0,
      "kind": "load",
      "length": 1,
      "params": Object {},
      "pathname": "/first",
      "pop": false,
      "query": Object {},
      "scene": "",
      "search": "",
      "state": Object {},
      "status": 200,
      "type": "FIRST",
      "url": "/first",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED_AGAIN",
    "universal": false,
    "url": "/redirected-again",
  },
  "title": "REDIRECTED_AGAIN",
}
`;

exports[`multiple redirects are honored: 1 - title 1`] = `"REDIRECTED_AGAIN"`;
